sys/kern/mutex.c

```diff
@@ -14,7 +14,7 @@ void mtx_init(mtx_t *m, lock_type_t type) {
   m->m_owner = NULL;
   m->m_count = 0;
   m->m_lockpt = NULL;
-  m->m_type = type;
+  m->m_type = type | LK_SLEEP;
```

Comment:
`LK_SLEEP` is defined as `0`, so this changes nothing.
Are logical operations on enums idiomatic C?
I also have an issue with the name `LK_SLEEP`: we don't actually go to sleep on a sleepqueue
when blocking on a mutex. I think `LK_BLOCK` or `LK_BLOCKING` is more appropriate.
The documentation in `sys/lock.h` should be updated.
Why not have something like this:
```c
typedef int lock_type_t;
#define LK_SLEEP 0x0
#define LK_SPIN 0x1
#define LK_TYPE 0x1
#define LK_RECURSE 0x2
```
The type of a lock can be extracted using `type & LK_TYPE`, and `LK_RECURSE` is now an ordinary flag.

Reply:
> `LK_SLEEP` is defined as 0, so this changes nothing.

Yes, but the same thing is done here:
https://github.com/cahirwpz/mimiker/blob/019d2787d696c24e1d6e14b579cd5eaeedbbd92f/include/sys/mutex.h#L26-L29
IMO it's more clear and errorproof in case of changing `LK_SLEEP` to some other value in the future.

> Are logical operations on enums idiomatic C?
> I also have an issue with the name `LK_SLEEP`:
> [...]

I agree that `lock_type_t` implementation may not be perfect, but this should probably be addressed in some other issue / PR? My PR is just fixing `*_init` functions so that spinlocks are not used as mutexes.